bulkio: Move target resolution code into backupbase package.

miretskiy opened the pull request:
A code movement refactoring which moves methods accessible
from outside of backupccl (currently ResolveTargetsToDescriptors) into
a separate backupbase package.  Because of various dependencies,
most supporting functions, including system schema backup/restore
information are also moved into backupbase.

This is a pure code movement to avoid dependency cycles between
backupccl and changefeedccl.  Now, all cross package dependencies
live in backupbase and changefeedbase respectively.

Release Notes: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59784)
<!-- Reviewable:end -->

Commit:
bulkio: Move target resolution code into backupbase package.

A code movement refactoring which moves methods accessible
from outside of backupccl (currently ResolveTargetsToDescriptors) into
a separate backupbase package. Because of various dependencies,
most supporting functions, including system schema backup/restore
information are also moved into backupbase.

This is a pure code movement to avoid dependency cycles between
backupccl and changefeedccl. Now, all cross package dependencies
live in backupbase and changefeedbase respectively.

Release Notes: None
